mcs/class/corlib/System.IO/FileStream.cs

```diff
@@ -188,6 +188,12 @@ internal FileStream (string path, FileMode mode, FileAccess access, FileShare sh
 				throw new ArgumentException ("Name has invalid chars");
 			}
 
+			// IsolatedStorage needs to keep the Name property to the default "[Unknown]"
+			if (!anonymous) {
+				path = Path.InsecureGetFullPath (path);
```

Comment:
Why is this inside the !anonymous if, shouldn't it apply to isolated storage as well?

Any particular reason why this was moved up?